src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -165,6 +172,15 @@ public TopicPartitionWriter(
       this.hiveUpdateFutures = hiveUpdateFutures;
       hivePartitions = new HashSet<>();
     }
+
+    if(rotateScheduleIntervalMs > 0) {
```

Comment:
I think it's probably better to move this check to `DataWriter` or even to `HdfsSinkTask` so that we can fail early.